flow/scripts/synth.tcl

```diff
@@ -56,7 +56,9 @@ if {[info exist ::env(PRESERVE_CELLS)]} {
 
 
 if {[info exist ::env(BLOCKS)]} {
-  hierarchy -check -top $::env(DESIGN_NAME)
+  if {![info exist ::env(PRESERVE_CELLS)]} {
+    hierarchy -check -top $::env(DESIGN_NAME)
+  }
```

Comment:
there is nothing wrong with calling hierarchy multiple times, so you don't need to wrap that with this statement.

Reply:
Done, updated.